core/src/impl/Kokkos_TaskNode.hpp

```diff
@@ -151,14 +151,19 @@ class ReferenceCountedBase {
   bool decrement_and_check_reference_count() {
     // TODO @tasking @memory_order DSH memory order
     auto old_count = Kokkos::atomic_fetch_add(&m_ref_count, -1);
+    Kokkos::memory_fence();
```

Comment:
Why memory fence rather than a desul atomic?

Reply:
I wanted to limit the number of wrapper atomics I implement.